.gitignore

```diff
@@ -0,0 +1,138 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+share/python-wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+*.manifest
+*.spec
+
+# Installer logs
+pip-log.txt
+pip-delete-this-directory.txt
+
+# Unit test / coverage reports
+htmlcov/
+.tox/
+.nox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+*.py,cover
+.hypothesis/
+.pytest_cache/
+cover/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
+*.log
+local_settings.py
+db.sqlite3
+db.sqlite3-journal
+
+# Flask stuff:
+instance/
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+.pybuilder/
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```

Comment:
I use pyenv and `.python-version` files, so we could uncomment this.

Reply:
What's the best practice with pyenv? I've been using virtualenv and my `.python-version` file simply says `growthviz`, the name of my virtualenv. Am I doing something wrong, or do we both agree on the same virtualenv name and what it means?

Reply:
I think there are a few common use patterns w/pyenv. I have both pyenv and pyenv-virtualenv installed via homebrew. My habit is to -- if, say, I have already run `pyenv install 3.9.2` -- to set up a new virtualenv using pyenv like this:

```bash
pyenv virtualenv 3.9.2 py39-growthviz
pyenv local py39-growthviz
```
The latter will create `.python-version` (w/contents: `py39-growthviz`), and with the bash_profile config specified at https://github.com/pyenv/pyenv#basic-github-checkout and https://github.com/pyenv/pyenv-virtualenv#installing-with-homebrew-for-macos-users, bash will always recognize which pyenv version I'm running by looking up the working dir hierarchy for a `.python-version`.

My goal here is that by gitignoring `.python-version`, you never need to know that my virtualenv is actually named `py39-growthviz`, or where the venv itself is located on my drive, nor I that yours is `growthviz`. I think it was pretty common before to always use a common name like `ENV` (hence the list in `.gitignore`), and I think I did something like that too before pyenv and pyenv-virtualenv came around.

So I think we can avoid agreeing on a name if you either use a common name for your venv which .gitignore already includes, or use a `.python-version` file like I'm doing. If you're already using pyenv and pyenv-virtualenv... it sounds like you are?... then just uncommenting the `.python-version` line should do us. Unless you're storing the virtualenv under the same dir as the repo, which might be best avoided with some additional pyenv config (e.g., `export PYENV_ROOT="$HOME/.pyenv"`).

Er, I should point out that for some reason I don't actually have 3.9.2 installed via pyenv yet, I don't think homebrew has picked it up yet, but I'm referring to 3.9.2 because there's a CVE on 3.9.1 and I don't want to document that I was using 3.9.1 at such a late date. :)

Reply:
🤦 sorry. Should have read your comment more closely. I'll uncomment the `.python-version` line in the `.gitignore`